### ISSUE TITLE
Failed to initialize NVML: GPU access blocked by the operating system

### ISSUE THREAD

**Issue by Marietto2008** [microsoft/WSL#10289]
### Windows Version

Windows 11 PRO Version 10.0.22000.2176

### WSL Version

Versione WSL: 1.3.11.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

6.1.38-microsoft-standard

### Distro Version

Ubuntu 22.10

### Repro Steps

Hello to everyone.

I tried to check my nvidia driver version within ubuntu 22.04 installed on the WSL2 subsystem installed on Windows 11,however, i got the error "Failed to initialize NVML: GPU access blocked by the operating system". Hope to get support from you on this issue.

On Windows 11 host os it works :
```

# nvidia-smi
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 536.40                 Driver Version: 536.40       CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                     TCC/WDDM  | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce GTX 1060 3GB  WDDM  | 00000000:01:00.0  On |                  N/A |
| 57%   41C    P2              25W / 120W |    710MiB /  3072MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 2080 Ti   WDDM  | 00000000:02:00.0 Off |                  N/A |
| 31%   40C    P8              21W / 250W |     74MiB / 11264MiB |      0%      
```

on Ubuntu 22.04 guest os it doesn't :

```
# nvidia-smi

Failed to initialize NVML: GPU access blocked by the operating system
Failed to properly shut down NVML: GPU access blocked by the operating system
```

BUT :

```
# nvcc -V

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Jun_13_19:16:58_PDT_2023
Cuda compilation tools, release 12.2, V12.2.91
Build cuda_12.2.r12.2/compiler.32965470_0


# python3 --version
Python 3.10.7
```

Within ubuntu 22.04 I have installed cuda-12.2 using the instructions below :

```
# wget https://developer.download.nvidia.com/compute/cuda/repos/wsl-ubuntu/x86_64/cuda-wsl-ubuntu.pin

# sudo mv cuda-wsl-ubuntu.pin /etc/apt/preferences.d/cuda-repository-pin-600

# wget https://developer.download.nvidia.com/compute/cuda/12.2.0/local_installers/cuda-repo-wsl-ubuntu-12-2-local_12.2.0-1_amd64.deb

# sudo dpkg -i cuda-repo-wsl-ubuntu-12-2-local_12.2.0-1_amd64.deb

# sudo cp /var/cuda-repo-wsl-ubuntu-12-2-local/cuda-*-keyring.gpg /usr/share/keyrings/

# sudo apt-get update

# sudo apt-get -y install nvidia-cuda-toolkit
```

My PC has 2 graphic cards :
  

>   NVIDIA GeForce GTX 1060 3GB
> 
>   NVIDIA GeForce RTX 2080 Ti

Below you can see the nvidia driver version installed on WIndows 11 for both the GPUs :

![cd](https://github.com/microsoft/WSL/assets/755339/dc7278ec-9e2a-4210-8a41-255ee1595a84)

I have also added the line below to /home/marietto/.bashrc and to /root/.bashrc :

`export PATH=/usr/local/cuda-12.2/bin${PATH:+:${PATH}}`

and I did :

```
# source /home/marietto/.bashrc
# source /root/.bashrc
```

### Expected Behavior

```
# nvidia-smi
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 536.40                 Driver Version: 536.40       CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                     TCC/WDDM  | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce GTX 1060 3GB  WDDM  | 00000000:01:00.0  On |                  N/A |
| 57%   41C    P2              25W / 120W |    710MiB /  3072MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA GeForce RTX 2080 Ti   WDDM  | 00000000:02:00.0 Off |                  N/A |
| 31%   40C    P8              21W / 250W |     74MiB / 11264MiB |      0%
```

### Actual Behavior

Failed to initialize NVML: GPU access blocked by the operating system

**Comment by Marietto2008** [comment]
Fixed installing the "5.15.90.3-microsoft-standard-WSL2" kernel.

**Comment by littleyeson** [comment]
> Fixed installing the "5.15.90.3-microsoft-standard-WSL2" kernel.

How to install?

**Comment by Marietto2008** [comment]
What about to use qemu + hyperV to virtualize Linux on top of Windows 11,instead of WSL2 ? I did that with FreeBSD as guest,but it can be done also with Linux :

https://forums.freebsd.org/threads/how-to-virtualize-freebsd-14-release-as-a-vm-on-top-of-windows-11-using-qemu-hyperv.93158/#post-652021
